connectors/slack/uninstall.js

```diff
@@ -5,8 +5,13 @@ This is the uninstallation logic of the Lifecycle Manager.
 const Sdk = require('@fusebit/add-on-sdk');
 
 module.exports = async (ctx) => {
-  // Destroy the Add-On Handler
+  let functionCtx = await Sdk.getFunctionDefinition(ctx);
+
+  await Sdk.deleteStorage(ctx, functionCtx.configuration);
```

Comment:
Casual suggestion - creation is storage -> function.  Deletion (usually) should be the mirror of this, so function first and then storage.

Reply:
This is how it used to be, but we switched it with an eye toward reentrancy. If the storage deletion fails, we'll fail the uninstall and then user can retry it in the portal in the hopes that the storage will eventually succeed and get cleaned up. If we were to delete the function first, there is no longer an entry point for the user to retry the deletion AND the secrets we need to be able to find the storage are now gone.